### DISCUSSION TITLE
Issue review for "Working with Views"

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#471] [Maintainer discussions]
I took an initial pass at creating design and backend issues for the "Working with Views" milestone. Please see:
- Meta-issue: #442 
- Design
  - #466
  - #454
  - #455
  - #456
- Backend
  - #467
  - #464
  - #465
  - #468
  - #469
  - #470

Views are an entirely new class of database object with a whole bunch of complexities, so I'd like to get feedback from @mathemancer and @pavish on implementation details and whether I've covered the necessary details in the design issues. I'm especially worried abut the view creation and editing flow – we need to convert lists of columns to queries with `JOIN`s and I'm not sure what we can and cannot automate easily.

Please review by end of day Monday, July 26.

@ghislaineguerin I'd also like your feedback on the design issues, but it's not as time-sensitive since I'm mainly focused on covering implementation details right now.

**Comment by mathemancer** [comment]
For an initial pass, I think the specification that we only make views from columns in tables joined by foreign key relationships is a good idea.  This makes automating joining fairly straightforward.

Editing records within a view is extremely complicated, and will require both time for definition and specification, as well as a quite complicated implementation involving both SQL and python.  We'll probably need bunches of sub-issues to track the work underlying that API change.

I commented on a couple of the specific issues with some more details.

**Reply by kgodey** [reply]
@mathemancer I'd like to understand editing records within a view better. I think we should have the conversation directly on that issue. I posted a follow up question.

I'm assuming everything else looks good/doable?

**Comment by pavish** [comment]
@kgodey The issues look good to me. I have the same concern as @mathemancer, regarding allowing editing records in views.

I think it will be quite time consuming to implement on the backend. Since this is a rare usecase, maybe we should reconsider having this for the MVP.

**Reply by kgodey** [reply]
@pavish I'm not so sure that it will be a rare use case. If we're trying to get people to use databases optimally, they'll be working with views very often, since their tables will be normalized and may not offer a great view of all the data they want to see. It would be ideal if they could update data wherever they were working.

As I mentioned in my response to @mathemancer, we should continue the conversation in #442. I'd like to understand the complexity involved a bit better before I make a decision on whether to remove it from the MVP scope. Postgres supports updates, inserts, and deletes directly into views in some cases, so I'm hoping that we can make use of that functionality without having to do anything too complicated on the backend.

**Comment by kgodey** [comment]
I'm calling this review complete, it looks like the issues as-is as reasonable. The only thing I made was a future issue:
- #504

**Comment by ghislaineguerin** [comment]
@kgodey I've reviewed the design issues and have no comments for now, they all look good to me.

**Reply by kgodey** [reply]
Thanks @ghislaineguerin!
